pandas/io/json/_json.py

```diff
@@ -716,6 +720,14 @@ def _get_data_from_filepath(self, filepath_or_buffer):
                 errors=self.encoding_errors,
             )
             filepath_or_buffer = self.handles.handle
+        elif (
```

Comment:
so IIRC we did do this quite awhile ago and in fact just did essentially `file_exists(filepath_or_buffer)` 

I believe this was reverted as it is not reliable and can potentially overflow the file system.

Reply:
@jreback Sounds very odd. How can file existence check go so wrong? Do you remember under what conditions? Was it a rare edge case? Couldn't find any old issue about this. Do you remember a keyword to search for?

Reply:
if its a string this can simply overflow the file system

Reply:
So would a check for string length before file_exists help?

```py
isinstance(filepath_or_buffer, str) and len(filepath_or_buffer) < 256 and ... # or whatever is safe
```

Reply:
`file_exists` tries to catch certain errors: 
https://github.com/pandas-dev/pandas/blob/cec066d221cc48a858fcae650d254b5392dd1247/pandas/io/common.py#L1095

Reply:
@twoertwein @jreback Maybe worth adding a test case for `read_json` that tries to load a very long file path?

Reply:
Seems like the concern @jreback raised is handled by file_exists. Any other reason not to merge this?

Reply:
ok this looks fine, can we do exactly that add a superlong json string here